hemi-viem-stake-actions/actions/wallet/stakeManager.ts

```diff
@@ -0,0 +1,102 @@
+import { type Address, type Client } from 'viem'
+import { writeContract } from 'viem/actions'
+
+import {
+  stakeManagerAbi,
+  stakeManagerAddresses,
+} from '../../contracts/stakeManager'
+
+/**
+ * Performs staking of an ERC-20 token by calling the `depositFor` function on the contract.
+ *
+ * @param {Client} client - The Viem client configured to interact with the blockchain.
+ * @param {Object} parameters - The transaction parameters for staking.
+ * @param {bigint} parameters.amount - The amount of tokens to be staked.
+ * @param {Address} parameters.from - The user's address initiating the staking.
+ * @param {Address} parameters.token - The ERC-20 token address being staked.
+ *
+ * @returns {Promise<void>} - The transaction sent to the blockchain.
+ */
+export function stakeERC20Token(
+  client: Client,
+  parameters: {
+    amount: bigint
+    from: Address
+    token: Address
+  },
+) {
+  const { amount, from, token } = parameters
+  return writeContract(client, {
+    abi: stakeManagerAbi,
+    account: from,
+    // @ts-expect-error: TS is complaining about client.chain!.id definition, but this works
+    address: stakeManagerAddresses[client.chain!.id],
+    args: [token, from, amount],
+    // @ts-expect-error: TS is complaining about client.chain definition, but this works
+    chain: client.chain,
+    functionName: 'depositFor',
+  })
+}
+
+/**
+ * Performs staking of native ETH by calling the `depositETHFor` function on the contract.
+ *
+ * @param {Client} client - The Viem client configured to interact with the blockchain.
+ * @param {Object} parameters - The transaction parameters for staking.
+ * @param {bigint} parameters.amount - The amount of tokens to be staked.
+ * @param {Address} parameters.from - The user's address initiating the staking.
+ *
+ * @returns {Promise<void>} - The transaction sent to the blockchain.
```

Comment:
Same here as in `stakeERC20Token`

```suggestion
 * @returns {Promise<Address>} - The transaction sent to the blockchain.
```

Reply:
Done.